server/go/subscription/manager.go

```diff
@@ -0,0 +1,67 @@
+package subscription
+
+import "sync"
+
+type Manager struct {
+	subscriptions map[string][]Subscriber
+	mutex         sync.Mutex
+}
+
+var managerInstance *Manager = nil
+
+func GetManager() *Manager {
+	if managerInstance == nil {
+		managerInstance = newManager()
+	}
+
+	return managerInstance
+}
+
+func newManager() *Manager {
+	return &Manager{
+		subscriptions: make(map[string][]Subscriber),
+		mutex:         sync.Mutex{},
+	}
+}
+
+func (m *Manager) Subscribe(resourceID string, subscriber Subscriber) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
```

Comment:
Does defer mean doing this after the rest of the code statements are executed?

Reply:
exactly. It is executed just before returning. [More details](https://go.dev/tour/flowcontrol/12)